nodebuilder/blobstream/types.go

```diff
@@ -0,0 +1,195 @@
+package blobstream
+
+import (
+	"fmt"
+
+	"github.com/celestiaorg/celestia-app/pkg/shares"
+
+	"github.com/celestiaorg/celestia-app/pkg/appconsts"
+
+	"github.com/celestiaorg/nmt"
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/tendermint/tendermint/crypto/merkle"
+
+	"github.com/celestiaorg/celestia-node/share"
+
+	"github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/types"
+)
+
+// ResultDataCommitment is the API response containing a data
+// commitment, aka data root tuple root.
+type ResultDataCommitment struct {
+	DataCommitment bytes.HexBytes `json:"data_commitment"`
+}
+
+// ResultDataRootInclusionProof is the API response containing the binary merkle
+// inclusion proof of a height to a data commitment.
+type ResultDataRootInclusionProof struct {
+	Proof merkle.Proof `json:"proof"`
+}
+
+// ResultShareProof is the API response that contains a ShareProof.
+// A share proof is a proof of a set of shares to the data root.
+type ResultShareProof struct {
+	ShareProof types.ShareProof `json:"share_proof"`
+}
+
+// ResultCommitmentProof is an API response that contains a CommitmentProof.
+// A commitment proof is a proof of a blob share commitment to the data root.
+type ResultCommitmentProof struct {
+	CommitmentProof CommitmentProof `json:"commitment_proof"`
+}
+
+// CommitmentProof is an inclusion proof of a commitment to the data root.
+// TODO: Ask reviewers if we need protobuf definitions for this
+type CommitmentProof struct {
```

Comment:
Feels like `CommitmentProof` could be pretty heavy it terms of memory usage. May be we should consider having a pointer to the receivers of its methods instead of value receivers?